### PR TITLE
KAFKA-3605: Return error if connector config includes mismatching connector name.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@Ishiihara @hachikuji @gwenshap Should be an easy review, and @gwenshap need you to commit :)

**Comment by Ishiihara** [comment]
LGTM

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java [hunk 1]
```diff
@@ -129,6 +130,9 @@ public Response putConnectorConfig(final @PathParam("connector") String connecto
                                        final @QueryParam("forward") Boolean forward,
                                        final Map<String, String> connectorConfig) throws Throwable {
         FutureCallback<Herder.Created<ConnectorInfo>> cb = new FutureCallback<>();
+        String includedName = connectorConfig.get(ConnectorConfig.NAME_CONFIG);
+        if (includedName != null && !includedName.equals(connector))
```
Comment (hachikuji) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java]: If the included name is null, then it looks like we depend on the Herder to set it. Would it make sense to move that here?

Comment (hachikuji) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java]: Actually it seems that `StandaloneHerder` is missing the check.

Comment (ewencp) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java]: Yeah, fair point, I realized we were a bit inconsistent. Will update to make the modifications in the REST API classes so we don't have to worry about it in the Herders.

**Comment by ewencp** [comment]
@gwenshap If you have a moment, have addressed review comments

**Comment by gwenshap** [comment]
LGTM

**Pull request closed without merging**